gateway/handler_error.go

```diff
@@ -128,23 +132,34 @@ func (e *ErrorHandler) HandleError(w http.ResponseWriter, r *http.Request, errMs
 			templateName = defaultTemplateName + "." + defaultTemplateFormat
 			tmpl = templates.Lookup(templateName)
 			w.Header().Set(headers.ContentType, defaultContentType)
+			response.Header.Set(headers.ContentType, defaultContentType)
+
 		}
 
 		//If the config option is not set or is false, add the header
 		if !e.Spec.GlobalConfig.HideGeneratorHeader {
 			w.Header().Add(headers.XGenerator, "tyk.io")
+			response.Header.Add(headers.XGenerator, "tyk.io")
 		}
 
 		// Close connections
 		if e.Spec.GlobalConfig.CloseConnections {
 			w.Header().Add(headers.Connection, "close")
+			response.Header.Add(headers.Connection, "close")
+
 		}
 
 		// If error is not customized write error in default way
 		if errMsg != errCustomBodyResponse.Error() {
 			w.WriteHeader(errCode)
+			response.StatusCode = errCode
+
 			apiError := APIError{template.HTML(template.JSEscapeString(errMsg))}
-			tmpl.Execute(w, &apiError)
+			var log bytes.Buffer
+
+			rsp := io.MultiWriter(w, &log)
+			tmpl.Execute(rsp, &apiError)
+			response.Body = ioutil.NopCloser(bytes.NewBufferString(log.String()))
```

Comment:
I think this will do 
```go
response.Body = ioutil.NopCloser(&log)
```
`bytes.Buffer` implements `io.Reader`

Reply:
done! thanks